modules/l4proxyprotocol/matcher.go

```diff
@@ -26,7 +24,7 @@ import (
 
 // https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
 var (
-	headerV1Prefix = []byte("PROX") // intentional to not include "Y", see match() function for details
+	headerV1Prefix = []byte("PROXY")
```

Comment:
Does this logic that was removed no longer apply?

```
// since 5th byte being null is valid in v2 header
// but is considered EOF by readers
```

Just wondering if this will break some things

Reply:
Yes it is no longer needed. I assumed it was needed because bufio has some special handling for text:

> // Package bufio implements buffered I/O. It wraps an io.Reader or io.Writer
> // object, creating another object (Reader or Writer) that also implements
> // the interface but provides buffering and some help for textual I/O.

but I did not dig deep into it.  
Instead I wrote a test to verify that io.ReadFull can read past a null byte just fine, see TestProxyProtocolMatchV2. The headerV2Prefix always contains 0x00 at the 5th position.